tests/Rector/Architecture/Factory/NewObjectToFactoryCreateRector/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Rector\Tests\Rector\Architecture\Factory\NewObjectToFactoryCreateRector\Fixture;
+
+use Rector\Tests\Rector\Architecture\Factory\NewObjectToFactoryCreateRector\Source\MyClass;
+
+final class SomeController
+{
+    public function default()
+    {
+        new MyClass('abcd');
```

Comment:
Could add a new line under this one with assign case?

```php
$myClass = new MyClass('abcd');
```

Reply:
Good catch! This causes error and i have to figure out and fix it 😉 👍 

Anyways, it is terrible for debugging with xdebug, i can not see what exact function call caused it and overall i have no idea 😄 

<img width="871" alt="screenshot 2019-01-24 22 58 10" src="https://user-images.githubusercontent.com/3995003/51711170-95c66300-202b-11e9-801a-42309e910792.png">

Reply:
After quick research i found out this is the problem:

```
$this->addNodeAfterNode($methodCallNode, $node);
$this->removeNode($node);
```

Because `removeNode()` only works if it is `new xx()` without variable assign.

I will figure out and dig more and let you know 😉 im looking for elegant way of replacing New with MethodCall node.

Reply:
Simple `return $methodCallNode;` did the trick 🎉